stub_uploader/metadata.py

```diff
@@ -174,6 +177,43 @@ def verify_typeshed_req(req: Requirement) -> None:
 }
 
 
+def validate_response(resp: requests.Response, req: Requirement) -> None:
+    if resp.status_code != 200:
+        raise InvalidRequires(
+            f"Expected dependency {req} to be accessible on PyPI, but got {resp.status_code}"
+        )
+
+
+def get_sdist_requires(
```

Comment:
Nit: "get" makes this function sound simpler than what it actually does. Maybe "extract"?

Also I think it would be safer to extract the sdist into an empty temporary directory than to where ever we currently are.